lib/Transforms/ConvertToCiphertextSemantics/ConvertToCiphertextSemantics.cpp

```diff
@@ -1064,6 +1064,22 @@ struct ConvertToCiphertextSemantics
 
     RewritePatternSet patterns(context);
     ConversionTarget target(*context);
+
+    // check for invalid types on values and bail out of the pass
+    auto isValidValue = [&](const Value value) -> LogicalResult {
+      auto contextualAttr = typeConverter.getContextualAttr(value);
+      if (failed(contextualAttr)) {
+        return failure();
+      }
+      Type result =
+          typeConverter.convertType(value.getType(), contextualAttr.value());
+      return result != nullptr ? success() : failure();
+    };
+    if (!walkAndValidateValues(module, isValidValue).succeeded()) {
+      module->emitError() << "invalid type in module";
```

Comment:
This looks OK, but could we possibly test this? Maybe add a test with an IR that uses some unsupported type like `memref` and tries to run this pass? Note you can use `-verify-diagnostics` to assert error messages are correctly generated.